content/blog/2018-03-13-weathercan.md

```diff
@@ -0,0 +1,500 @@
+  filter(end >= 2017)
+```
+
+    ## # A tibble: 2 x 9
+    ##   station_name station_id   lat   lon  elev interval start   end distance
+    ##   <chr>        <fct>      <dbl> <dbl> <dbl> <chr>    <int> <int>    <dbl>
+    ## 1 KAMLOOPS AUT 42203       50.7  -120   345 hour      2006  2018     6.18
+    ## 2 KAMLOOPS A   51423       50.7  -120   345 hour      2013  2018     6.79
+
+Next, download the weather data:
+
+``` r
+kam <- weather_dl(station_ids = "42203", start = min(f$date), end = max(f$date)) %>%
+  select(time, temp)
+kam
+```
+
+    ## # A tibble: 120 x 2
+    ##    time                 temp
+    ##  * <dttm>              <dbl>
+    ##  1 2017-01-06 00:00:00 -11.2
+    ##  2 2017-01-06 01:00:00 -10.7
+    ##  3 2017-01-06 02:00:00 -10.8
+    ##  4 2017-01-06 03:00:00 -10.4
+    ##  5 2017-01-06 04:00:00 -14.0
+    ##  6 2017-01-06 05:00:00 -14.5
+    ##  7 2017-01-06 06:00:00 -11.1
+    ##  8 2017-01-06 07:00:00 -10.9
+    ##  9 2017-01-06 08:00:00 -11.7
+    ## 10 2017-01-06 09:00:00 -12.7
+    ## # ... with 110 more rows
+
+Finally, we use the `weather_interp` function from `weathercan` to perform a linear interpolation and add this temperature data to our feeder observations:
+
+``` r
+f_temp <- weather_interp(f, kam, cols = "temp")
+glimpse(f_temp)
+```
+
+    ## Observations: 2,273
+    ## Variables: 11
+    ## $ animal_id <fct> 062000042E, 062000042E, 062000042E, 062000042E, 0620...
+    ## $ date      <date> 2017-01-06, 2017-01-06, 2017-01-06, 2017-01-06, 201...
+    ## $ time      <dttm> 2017-01-06 11:12:49, 2017-01-06 11:14:09, 2017-01-0...
+    ## $ logger_id <fct> 1500, 1500, 1500, 1500, 1500, 1500, 1500, 1500, 1500...
+    ## $ species   <chr> "Mountain Chickadee", "Mountain Chickadee", "Mountai...
+    ## $ age       <chr> "AHY", "AHY", "AHY", "AHY", "AHY", "AHY", "AHY", "AH...
+    ## $ sex       <chr> "U", "U", "U", "U", "U", "U", "U", "U", "U", "U", "U...
+    ## $ site_name <chr> "Kamloops, BC", "Kamloops, BC", "Kamloops, BC", "Kam...
+    ## $ lon       <dbl> -120.3658, -120.3658, -120.3658, -120.3658, -120.365...
+    ## $ lat       <dbl> 50.67057, 50.67057, 50.67057, 50.67057, 50.67057, 50...
+    ## $ temp      <dbl> -9.793194, -9.782083, -9.775417, -9.760556, -9.74513...
+
+To illustrate how this works, let's take a look at the temperature measures downloaded with `weathercan` (large black dots), vs. the interpolated values now stored with the feeder data (small red dots).
+
+``` r
+ggplot(data = f_temp[1:25,], aes(x = time, y = temp)) +
+  theme_bw() + 
+  theme(legend.position = c(0.2, 0.8)) +
+  geom_point(data = kam[12:17,], aes(colour = "weathercan temperature"), size = 4) +
+  geom_point(aes(colour = "Interpolated temperature"), size = 1) +
+  scale_colour_manual(name = "", values = c("red", "black")) +
+  labs(x = "Time", y = "Temperature (C)")
+```
+
+<center>
+<img src = "/img/blog-images/2018-03-13-weathercan/interpolate-1.png" style = "width: 100%">
+</center>
+<p>
```

Comment:
I like how the interpolation section explains a cool thing you can do, however, like the "Simple Example" above, it needs a strong opening couple of lines saying what question you are asking (about temp and birds) and therefore, what data you will combine.